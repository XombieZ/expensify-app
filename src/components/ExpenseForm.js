import React from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";

const now = moment();
console.log(now);

export default class ExpenseForm extends React.Component {
  state = {
    description: "",
    note: "",
    amount: "",
    createdAt: moment(),
    calendarFocused: false,
    error: ""
  };

  constructor(props) {
    super(props);

    if (props.expenseToEdit) {
      this.state = {
        ...this.state,
        description: props.expenseToEdit.description,
        note: props.expenseToEdit.note,
        amount: (props.expenseToEdit.amount / 100).toString(),
        createdAt: moment(props.expenseToEdit.createdAt)
      };
    }
  }

  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };

  onNoteChange = e => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };

  onAmountChange = e => {
    const amount = e.target.value;

    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };

  onDateChange = createdAt => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };

  onFocusChange = ({ focused }) =>
    this.setState(() => ({ calendarFocused: focused }));

  onSubmit = e => {
    e.preventDefault();

    if (!this.state.description || !this.state.amount) {
      // Set error
      this.setState(() => ({
        error: "Please provide description and amount!"
      }));
    } else {
      // Clear error
      this.setState(() => ({ error: "" }));
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100, // *100 => calculating in cents
        createdAt: this.state.createdAt.valueOf(), // timestamp in milliseconds moment().valueOf()
        note: this.state.note
      });
    }
  };

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea
            value={this.state.note}
            onChange={this.onNoteChange}
            placeholder="Add a note for your expense (optional)"
          />
          <button>Add Expense</button>
        </form>
      </div>
    );
  }
}
