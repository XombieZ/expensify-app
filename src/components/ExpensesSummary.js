import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import getVisibleExpenses from "../selectors/expenses";
import selectExpensesTotal from "../selectors/expenses-total";

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
  const sufix = expenseCount > 1 ? "s" : "";
  const text = `Viewing ${expenseCount} expense${sufix} totalling $${numeral(
    expenseTotal / 100
  ).format("0.00")}`;

  return <div>{expenseTotal > 0 && <p>{text}</p>}</div>;
};

const mapStateToProps = state => {
  return {
    expenseCount: getVisibleExpenses(state.expenses, state.filters).length,
    expenseTotal: selectExpensesTotal(
      getVisibleExpenses(state.expenses, state.filters)
    )
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
