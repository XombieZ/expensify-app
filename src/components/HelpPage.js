import React from "react";

const HelpPage = () => {
  return <div>This is from Help page</div>;
};

export default HelpPage;
