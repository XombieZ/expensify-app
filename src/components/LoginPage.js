import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";
import { facebookAuthProvider, googleAuthProvider } from "../firebase/firebase";

export class LoginPage extends React.Component {
  render() {
    return (
      <div className="box-layout">
        <div className="box-layout__box">
          <h1 className="box-layout__title">Expensify</h1>
          <p>It's time to get your expenses under control. </p>
          <button className="button" onClick={this.props.startLoginWithGoogle}>
            Login with Google
          </button>
          <button
            className="button"
            onClick={this.props.startLoginWithFacebook}
          >
            Login with Facebook
          </button>
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  startLoginWithGoogle: () => dispatch(startLogin(googleAuthProvider)),
  startLoginWithFacebook: () => dispatch(startLogin(facebookAuthProvider))
});

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
