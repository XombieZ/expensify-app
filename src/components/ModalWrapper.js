import React from "react";

class ModalWrapper extends React.Component {
  render() {
    return (
      <div
        className={
          !this.props.isHidden
            ? "modal-wrapper modal-wrapper--visible"
            : "modal-wrapper modal-wrapper--hidden"
        }
      >
        <div className="modal-wrapper-content">{this.props.children}</div>
      </div>
    );
  }
}

export default ModalWrapper;
