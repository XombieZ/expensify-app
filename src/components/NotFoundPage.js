import React from "react";
import { Link } from "react-router-dom";

const NotFoundPage = () => {
  return <div>404!</div>;
};

export default NotFoundPage;
