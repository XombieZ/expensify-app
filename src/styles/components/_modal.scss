.modal-wrapper {
  background: rgba(0, 0, 0, 0.5);
  height: 100vh;
  left: 0px;
  position: absolute;
  top: 0px;
  width: 100vw;
  z-index: 1;
}

.modal-wrapper-content {
  opacity: 1;
  z-index: 1000;
}

.modal-wrapper--hidden {
  display: none;
}

.modal-wrapper--visible {
  align-items: center;

  display: flex;
  justify-content: center;
}

.modal-confirmation {
  align-items: center;
  background: $off-white;
  border-radius: 5px;
  display: flex;
  height: 20rem;
  justify-content: center;
  flex-direction: column;
  padding: 1rem;
  width: 25rem;
}

.modal-confirmation > h3 {
  text-align: center;
}

.modal-confirmation-inputs {
  display: flex;
  justify-content: space-between;
  width: 60%;
}
