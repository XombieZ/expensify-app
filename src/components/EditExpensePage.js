import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import ModalWrapper from "./ModalWrapper";
import ConfirmationModal from "./modals/ConfirmationModal";
import { startEditExpense, startRemoveExpense } from "../actions/expenses";

export class EditExpensePage extends React.Component {
  state = { isRemoveModalHidden: true, isSaveModalHidden: true };

  setRemoveModalHidden = isHidden =>
    this.setState({ isRemoveModalHidden: isHidden });

  onSubmit = expense => {
    this.props.startEditExpense(this.props.expenseToEdit.id, expense);
    this.props.history.push("/dashboard"); // go to index page
  };

  onRemoveModalConfirm = () => {
    this.setRemoveModalHidden(true);
    this.props.startRemoveExpense({ id: this.props.expenseToEdit.id });
    this.props.history.push("/dashboard"); // go to index page
  };

  onRemoveModalCancel = () => {
    this.setState({ isRemoveModalHidden: true });
    console.log("Modal: cancel");
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expenseToEdit={this.props.expenseToEdit}
            onSubmit={this.onSubmit}
          />
          <button
            className="button button--secondary"
            onClick={() => this.setRemoveModalHidden(false)}
          >
            Remove Expense
          </button>
        </div>
        <ModalWrapper isHidden={this.state.isRemoveModalHidden}>
          <ConfirmationModal
            message="Are you sure you want to delete current expense ?"
            onConfirm={this.onRemoveModalConfirm}
            onCancel={this.onRemoveModalCancel}
          />
        </ModalWrapper>
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    expenseToEdit: state.expenses.find(
      expense => expense.id === ownProps.match.params.id
    )
  };
};

const mapDispatchToProps = dispatch => {
  return {
    startEditExpense: (id, updates) => dispatch(startEditExpense(id, updates)),
    startRemoveExpense: id => dispatch(startRemoveExpense(id))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditExpensePage);
