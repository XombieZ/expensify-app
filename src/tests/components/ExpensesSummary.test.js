import React from "react";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import { shallow } from "enzyme";

test("should render correctly with 1 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expenseTotal={20300} />
  );

  expect(wrapper).toMatchSnapshot();
});

test("should render correctly with 2 expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={2} expenseTotal={13034} />
  );
  u;

  expect(wrapper).toMatchSnapshot();
});
