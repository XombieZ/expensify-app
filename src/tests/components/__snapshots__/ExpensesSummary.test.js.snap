// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with 1 expense 1`] = `
<div>
  <p>
    Viewing 1 expense totalling $203.00
  </p>
</div>
`;

exports[`should render correctly with 2 expenses 1`] = `
<div>
  <p>
    Viewing 2 expenses totalling $130.34
  </p>
</div>
`;
